sbi/inference/posteriors/base_posterior.py

```diff
@@ -980,16 +980,48 @@ def __setstate__(self, state_dict: Dict):
         """
         Sets the state when being loaded from pickle.
 
+        For developers: for any new attribute added to `NeuralPosterior`, we have to 
+        add an entry here using `check_warn_and_setstate()`.
+
         Args:
             state_dict: State to be restored.
         """
-        if "_device" not in state_dict.keys():
-            state_dict["_device"] = "cpu"
-            warn(
+
+        # In the beginning, the warning message is empty.
+        warning_msg = ""
+
+        state_dict, warning_msg = check_warn_and_setstate(
+            state_dict, "_device", "cpu", warning_msg
+        )
+
+        if state_dict["_x"] is not None:
+            state_dict, warning_msg = check_warn_and_setstate(
+                state_dict, "_num_iid_trials", state_dict["_x"].shape[0], warning_msg
+            )
+        else:
+            state_dict, warning_msg = check_warn_and_setstate(
+                state_dict, "_num_iid_trials", None, warning_msg
+            )
+
+        state_dict, warning_msg = check_warn_and_setstate(
+            state_dict, "_iid_methods", ["snle", "snre_a", "snre_b"], warning_msg
+        )
+
+        state_dict, warning_msg = check_warn_and_setstate(
+            state_dict,
+            "_allow_iid_x",
+            state_dict["_method_family"] in state_dict["_iid_methods"],
+            warning_msg,
+        )
+
+        if warning_msg:
+            warning_description = (
                 "You had saved the posterior under an older version of `sbi`. To make "
```

Comment:
"You saved"

Reply:
I prefer `had saved` here because it emphasizes that the posterior had been saved before it now got loaded